spec/models/post_spec.rb

```diff
@@ -2,7 +2,50 @@
 
 RSpec.describe Post, type: :model do
   it { should belong_to :author }
+  it { should have_many(:tags).through(:taggings) }
   it { should validate_presence_of :body }
   it { should validate_presence_of :title }
   it { should validate_presence_of :author }
+
+#   describe "#all_tags" do
+#     it "should validate associated_models" do
+#       post = Fabricate.build(:post)
+#       post.all_tags = "!#@$  1!@\#$ASDF "
+#       post.all_tags.should have_error("Tags should be separated by commas and should not have special characters")
+#     end
+#   end
+
```

Comment:
You shouldn't commit commented out code.  At the very least, this should be marked as pending.